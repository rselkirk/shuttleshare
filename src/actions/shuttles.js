import uuid from 'uuid';

// ADD_SHUTTLE
export const addShuttle = (
  { origin = '',
    destination = '',
    date = '',
    time = '',
    spots = '',
    cost = ''
  } = {}
) => ({
  type: 'ADD_SHUTTLE',
  shuttle: {
    id: uuid(),
    origin,
    destination,
    date,
    time,
    spots,
    cost
  }
});

// REMOVE_SHUTTLE
export const removeShuttle = ({ id } = {}) => ({
  type: 'REMOVE_SHUTTLE',
  id
});

// EDIT_SHUTTLE
export const editShuttle = (id, updates) => ({
  type: 'EDIT_SHUTTLE',
  id,
  updates
});