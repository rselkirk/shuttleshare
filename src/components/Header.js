import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = (props) => {
  return (
    <div>
      <h1>{props.title}</h1>
      <h2>{props.subtitle}</h2>
      <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
      <NavLink to="/create" activeClassName="is-active">Add Shuttle</NavLink>
      <NavLink to="/help" activeClassName="is-active">Help</NavLink>

    </div>
  );
};

Header.defaultProps = {
  title: 'ShuttleShare',
  subtitle: 'Rossland rideshare for people and bikes'
};

export default Header;