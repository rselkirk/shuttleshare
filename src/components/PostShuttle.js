import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';

// const useStyles = makeStyles(theme => ({
//   root: {
//     display: 'flex',
//     flexWrap: 'wrap',
//   },
//   formControl: {
//     margin: theme.spacing(1),
//     minWidth: 120,
//   },
//   selectEmpty: {
//     marginTop: theme.spacing(2),
//   },
// }));

// const classes = useStyles();


export default class PostShuttle extends React.Component {
  state = {
    error: undefined,
    from: '',
    to: '',
    date: '',
    time: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };
  
  handlePostShuttle = (e) => {
    e.preventDefault();

    const shuttle = { 
      "from": this.state.from,
      "to": this.state.to,
      "date": this.state.date,
      "time": this.state.time,
      "spots": e.target.elements.spots.value.trim(),
      "cost": e.target.elements.cost.value.trim(),
    };

    const error = this.props.handlePostShuttle(shuttle);

    this.setState(() => ({ error }));

    if (!error) {
      this.state.from = '';
      this.state.to = '';
      this.state.date = '';
      this.state.time = '';
      this.state.spots = '';
      this.state.cost = '';
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.handlePostShuttle}>
          <FormControl>
            <InputLabel htmlFor="from">From</InputLabel>
            <Select
              onChange={this.handleChange}
              value={this.state.from}
              inputProps={{
                name: 'from'
              }}
            >
              <MenuItem value="Rossland">Rossland</MenuItem>
              <MenuItem value="Neptune Creek">Neptune Creek</MenuItem>
              <MenuItem value="Malde Creek">Malde Creek</MenuItem>
            </Select>
          </FormControl>
          <FormControl>
            <InputLabel htmlFor="to">To</InputLabel>
            <Select
              onChange={this.handleChange}
              value={this.state.to}
              inputProps={{
                name: 'to'
              }}
            >
              <MenuItem value="Rossland">Rossland</MenuItem>
              <MenuItem value="Neptune Creek">Neptune Creek</MenuItem>
              <MenuItem value="Malde Creek">Malde Creek</MenuItem>
            </Select>
          </FormControl>
          <FormControl>
            <InputLabel htmlFor="date">Date</InputLabel>
            <Select
              onChange={this.handleChange}
              value={this.state.date}
              inputProps={{
                name: 'date'
              }}
            >
              <MenuItem value="Aug 1">Aug 1</MenuItem>
              <MenuItem value="Aug 2">Aug 2</MenuItem>
            </Select>
          </FormControl>
          <FormControl>
            <InputLabel htmlFor="time">Time</InputLabel>
            <Select
              onChange={this.handleChange}
              value={this.state.time}
              inputProps={{
                name: 'time'
              }}
            >
              <MenuItem value="3:00 pm">3:00 pm</MenuItem>
              <MenuItem value="4:00 pm">4:00 pm</MenuItem>
            </Select>
          </FormControl>
          <FormControl>
            <TextField
              id="spots"
              value={this.state.spots}
              label="Spots"
              margin="normal"
            />
          </FormControl>
          <FormControl>
            <TextField
              id="cost"
              value={this.state.cost}
              label="Cost"
              margin="normal"
            />
          </FormControl>
          <Button type="submit">Post Shuttle</Button>
          </form>
      </div>
    );
  }
}