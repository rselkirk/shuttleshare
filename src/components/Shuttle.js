import React from 'react';

const Shuttle = (props) => {
  return (
    <div>
      {props.shuttleText}
      <button onClick={(e) => {
        props.handleDeleteShuttle(props.shuttleText);
      }}
      >
        Remove
      </button>
    </div>
  );
}

export default Shuttle;