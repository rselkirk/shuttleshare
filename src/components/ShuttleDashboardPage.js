import React from 'react';
import ShuttleList from './ShuttleList';

const ShuttleDashboardPage = () => (
  <div>
    <ShuttleList />
  </div>
);

export default ShuttleDashboardPage;