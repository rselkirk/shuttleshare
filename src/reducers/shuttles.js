// Shuttles reducer

const shuttlesReducerDefaultState = [];

export default (state = shuttlesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_SHUTTLE':
      console.log("shuttle added")
      console.log(action.shuttle);
      return [
        ...state,
        action.shuttle
      ];
    case 'REMOVE_SHUTTLE':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_SHUTTLE':
      return state.map((shuttle) => {
        if (shuttle.id === action.id) {
          return {
            ...shuttle,
            ...action.updates
          };
        } else {
          return shuttle;
        };
      });
    default:
      return state;
  }
};
