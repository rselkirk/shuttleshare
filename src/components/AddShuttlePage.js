import React from 'react';
import { connect } from 'react-redux';
import ShuttleForm from './ShuttleForm';
import { addShuttle } from '../actions/shuttles';

export class AddShuttlePage extends React.Component {
  onSubmit = (shuttle) => {
    this.props.addShuttle(shuttle);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Shuttle</h1>
        <ShuttleForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addShuttle: (shuttle) => dispatch(addShuttle(shuttle))
});

export default connect(undefined, mapDispatchToProps)(AddShuttlePage);