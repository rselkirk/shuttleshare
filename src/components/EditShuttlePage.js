import React from 'react';
import { connect } from 'react-redux';
import ShuttleForm from './ShuttleForm';
import { editShuttle, removeShuttle } from '../actions/shuttles';

const EditShuttlePage = (props) => {
  return (
    <div>
      <ShuttleForm
        shuttle={props.shuttle}
        onSubmit={(shuttle) => {
          props.dispatch(editShuttle(props.shuttle.id, shuttle));
          props.history.push('/');
          console.log('updated', shuttle);
        }}
      />
      <button onClick={(shuttle) => {
        props.dispatch(removeShuttle({ id: props.shuttle.id }));
        props.history.push('/');
      }}>Remove</button>
    </div>
  );
};

const mapStatetoProps = (state, props) => {
  return {
    shuttle: state.shuttle.find((shuttle) => shuttle.id === props.match.params.id)
  };
};

export default connect(mapStatetoProps)(EditShuttlePage);
