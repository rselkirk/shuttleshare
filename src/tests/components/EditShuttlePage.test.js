import React from 'react';
// import { shallow } from 'enzyme';
import { createShallow } from '@material-ui/core/test-utils';
import { EditShuttlePage } from '../../components/EditShuttlePage';
import shuttles from '../fixtures/shuttles';

let editShuttle, removeShuttle, history, shallow, wrapper;

beforeEach(() => {
  editShuttle = jest.fn();
  removeShuttle = jest.fn();
  history = { push: jest.fn() };
  shallow = createShallow();
  wrapper = shallow(
    <EditShuttlePage
      editShuttle={editShuttle}
      removeShuttle={removeShuttle}
      history={history}
      shuttle={shuttles[2]}
    />
  );
});

test('should render the EditShuttlePage', () => {
  expect(wrapper).toMatchSnapshot();
});

// test('should handle EditShuttle', () => {
//   wrapper.find('ShuttleForm').prop('onSubmit')(shuttles[2]);
//   expect(history.push).toHaveBeenLastCalledWith('/');
//   expect(editShuttle).toHaveBeenLastCalledWith(shuttles[2].id, shuttles[2]);
// });

test('should handle removeShuttle', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeShuttle).toHaveBeenLastCalledWith({
    id: shuttles[2].id
  });
});