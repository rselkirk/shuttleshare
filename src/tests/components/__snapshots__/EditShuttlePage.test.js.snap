// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the EditShuttlePage 1`] = `
<div>
  <WithStyles(ShuttleForm)
    onSubmit={[Function]}
    shuttle={
      Object {
        "cost": 550,
        "date": "1970-01-01T00:05:00.000Z",
        "destination": "Rossland",
        "id": 3,
        "origin": "Malde Creek",
        "spots": 4,
        "time": "1970-01-01T00:05:00.000Z",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
